### PR TITLE
Feature: Alter the delay-strategy from config.

### PR DESCRIPTION
**Pull request by adm-bome**
- manage the delay-strategy from config.
 - add backoff-strategy on delayed messages.
 - add prioritization of delayed messages.

### CODE REVIEW

**Comment by adm-bome** [comment]
#232

File: config/rabbitmq.php [hunk 1]
```diff
@@ -24,6 +24,46 @@
      */
     'factory_class' => Enqueue\AmqpLib\AmqpConnectionFactory::class,
 
+    /*
```
Comment (vyuldashev) [inline config/rabbitmq.php]: Lets put this whole block as a last key of config

**Comment by vyuldashev** [comment]
Its a huge one. The feature is interesting. I need some time to go through it. Also, tests would be useful.

**Comment by vyuldashev** [comment]
How will this strategy work with this: https://github.com/vyuldashev/laravel-queue-rabbitmq/blob/master/src/Queue/RabbitMQQueue.php#L117?

File: config/rabbitmq.php (hunk 1 above)
Comment (adm-bome) [reply config/rabbitmq.php]: Ok. I will move this to the end

**Comment by adm-bome** [comment]
As intended.
The delay is in seconds in laravel. So when there was an delay set in an command/worker, milliseconds are set into the Producer. This feature does not change this behavior but follows it.


This Feature only adds the ability to choose between DelayStrategy's, as stated in the config. The way the code currently works, was not changed. The feature hooks into this.

Giving developers the ability to:
- do nothing, and use the thirth party RabbitMqDlxDelayStrategy (default).
- choose there own DelayStrategy (when they write one).
- choose the already available `DlxDelayStrategy` in this repo (added).

In case of the `DlxDelayStrategy` developers can:
- add an **optional** BackoffStrategy.
- configure an **optional** Prioritization for messages when they where delayed multiple times.

> When RabbitMq is going to handle multiple messages with an delay, and the delay is increasing by the times the messages were delayed (backoff), then the delayed messages can not stay into one DLX queue, because rabbitMQ does not handle this correctly (by design). (also see: the link in #232)

**Pull request closed without merging**